src/module_compilation.ml

```diff
@@ -159,27 +166,12 @@ let build_cm cctx ~dep_graphs ~precompiled_cmi ~cm_kind (m : Module.t) =
               ; As extra_args
               ; if dynlink || cm_kind <> Cmx then As [] else A "-nodynlink"
               ; A "-no-alias-deps"; opaque_arg
-              ; (match CC.alias_module cctx with
-                 | None -> S []
-                 | Some (m : Module.t) ->
-                   As ["-open"; Module.Name.to_string (Module.name m)])
+              ; opens modules m
               ; As (match stdlib with
                   | None -> []
-                  | Some { Dune_file.Library.Stdlib.modules_before_stdlib; _ } ->
-                    let flags = ["-nopervasives"; "-nostdlib"] in
-                    if Module.Name.Set.mem modules_before_stdlib
-                         (Module.name m) then
-                      flags
-                    else
-                      match CC.lib_interface_module cctx with
-                      | None -> flags
-                      | Some m' ->
-                        (* See comment in [Dune_file.Stdlib]. *)
-                        if Module.name m = Module.name m' then
-                          "-w" :: "-49" :: flags
-                        else
-                          "-open" :: Module.Name.to_string (Module.name m')
-                          :: flags)
+                  | Some _ ->
+                    (* XXX why aren't these just normal library flags? *)
```

Comment:
Do you mean why we don't write `(flags -no-pervasives -nostdlib :standard)` in the `dune` file? It's just that we always need these flags for the stdlib, so it feels redundant to write them.

Reply:
Okay. It just seems like if we can do with a more minimal feature set then we should do it. It's also easier to understand and change for users who aren't completely familiar with dune.

Reply:
I guess. No objection to removing that, but the compiler `stdlib/dune`'s file needs to be adapted first.

Reply:
Hmm, I suppose this needs to be co-ordinated somehow. Make the PR to the compiler source after the release?